packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -366,9 +429,12 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
     if (value.text != _lastFieldText) {
       shouldUpdateOptions = true;
       _onChangedCallId += 1;
+      _isLoadingOptions = true;
+      _updateOptionsViewVisibility();
     }
     _lastFieldText = value.text;
     final int callId = _onChangedCallId;
+
     final Iterable<T> options = await widget.optionsBuilder(value);
```

Comment:
probably not related to this PR, but I'm curious why do we still need to run the optionsBuilder if shouldUpdateOptions is already false?

Reply:
Since the `optionsBuilder` uses `TextEditingValue` as input, we agreed upon letting the `optionsBuilder` run on non textual changes, but the we don't want to run the update options logic in that case, thats why i had to add this boolean, we wont need this boolean once we deprecate `optionsBuilder` and use `String` text instead of `TextEditingValue`.
you might wanna read the conversation from here https://github.com/flutter/flutter/pull/150776#issuecomment-2322035319

Reply:
Ah so `optionsBuilder` is still run in case there are side effects in it. Makes sense to me.